src/c++/perf_analyzer/load_manager.cc

```diff
@@ -473,9 +473,15 @@ LoadManager::PrepareInfer(InferContext* ctx)
         input.second.datatype_));
     ctx->inputs_.push_back(infer_input);
 
+    data_ptr = nullptr;
```

Comment:
Why is this needed if data_ptr was already initialized as nullptr in L457?

Reply:
`GetInputData()` does not explicitly set `data_ptr` to `nullptr` when no data is found, so `data_ptr` could contain the address of a previous iteration's found data, breaking the if condition below, which relies on `data_ptr` being `nullptr` to indicate no data was found for current iteration. This is why I pre-set `data_ptr` to `null_ptr` right here in each iteration. The initialization above outside the loop is just for general readability (all variables should be initialized).